### DISCUSSION TITLE
Raspberry Pi 2 revision 1.2 (BCM2837)

### DISCUSSION THREAD

**Discussion by diyelectromusic** [rsta2/circle#407] [General]
With regards to the following:
"There are also Raspberry Pi 2 versions and the Raspberry Pi Zero 2, which are based on the BCM2837 SoC. These Raspberry Pi versions can be used too (with RASPPI = 3)."

What config.txt settings would be required to allow this to successfully boot?  Do we need to set something up for [pi2] for 64-bit mode and use of kernel8.img?

Or would it be better to build a kernel8-32.img 32-bit kernel for it instead and use that?

Thanks,
Kevin

**Comment by rsta2** [comment]
> "There are also Raspberry Pi 2 versions and the Raspberry Pi Zero 2, which are based on the BCM2837 SoC. These Raspberry Pi versions can be used too (with RASPPI = 3)."

This note is from the AArch64 section. You can also build for the RPi 2B with BCM2837 SoC with `AARCH = 32` and `RASPPI = 2` or `RASPPI = 3`.

If you would set in *config.txt*

```
[pi2]
kernel=kernel8.img
```

this would work on the RPi 2B with BCM2837 SoC, but not on the RPi 2B with BCM2836 SoC. Unfortunately you have no chance to distinguish between the two variants of the RPi 2B in *config.txt*. `[pi2]` is valid for both variants.

So I would recommend to use AArch32 for the RPi 2B (build with `AARCH = 32` and `RASPPI = 2`):

```
[pi2]
kernel=kernel7.img
```

This will work for both variants.

Rene

**Reply by diyelectromusic** [reply]
Thanks - I think that is pretty much my conclusion.  I'll see what I can do.

I'm struggling to understand not just the logic of a "revision" of the Pi 2 with a different main CPU but then also not having the ability to distinguish between them in the config.txt file!

Kevin
